### PR TITLE
bls crate

### LINKED ISSUE #49: bls crate bug
There is a bug when we use bls key to sign  the message and use the signature for [writing] (https://github.com/supernovahs/eigen-rs/blob/29e17a6cf74f2df8972166a5f168702f13b36d30/crates/chainio/clients/avsregistry/src/writer.rs#L119)

Itried using this https://github.com/Layr-Labs/rust-bls-bn254 but, the error was same. Afk suggested maybe the way we generate the public key is wrong.

the mangata finance team made a PR with their bls keystore , and bls utils . Yet to integrate it , to check if it works.
https://github.com/supernovahs/eigen-rs/pull/42

Error : Scalar mul error using tenderly tracing

https://holesky.etherscan.io/tx/0x7e682b8b0eb5ae1735db0661d71aff5db128d7399d042e5408e82f5ea67790c8

### PR DESCRIPTION
**Pull request by supernovahs**
closes #49

### CODE REVIEW

**Comment by supernovahs** [comment]
- Commented out some crates  in services, as they will be integrated with  the new bls code and tested
- Will change the example https://github.com/Layr-Labs/eigensdk-rs/blob/supernovahs/bls/examples/avsregistry-write/examples/register_operator_in_quorum_with_avs_registry_coordinator.rs  from current holesky to anvil since an operator can only register once , will do it in a separate PR

**Comment by dralves** [comment]
needs to resolve a few conflicts

**Comment by supernovahs** [comment]
@dralves  fixed the conflicts

File: crates/chainio/clients/avsregistry/src/writer.rs [hunk 1]
```diff
@@ -149,85 +141,73 @@ impl AvsRegistryChainWriter {
...
+                        },
+                        pubkeyG2: RegistryG2Point {
+                            X: pub_key_reg_params.pubkeyG2.X,
+                            Y: pub_key_reg_params.pubkeyG2.Y,
+                        },
+                    },
+                    operator_signature_with_salt_and_expiry,
+                );
+
+                let tx_call = contract_call.gas(2000000);
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/writer.rs:202]: pull a constant with proper naming

File: crates/chainio/clients/avsregistry/src/writer.rs [hunk 2]
```diff
@@ -149,85 +141,73 @@ impl AvsRegistryChainWriter {
...
+                let tx_result = tx_call.send().await;
+
+                match tx_result {
+                    Ok(tx) => {
+                        info!(tx_hash = ?tx,"succesfully deregistered operator with the AVS's registry coordinator" );
```
Comment (dralves) [inline crates/chainio/clients/avsregistry/src/writer.rs:207]: nit: capitalize S

File: crates/chainio/clients/elcontracts/src/writer.rs [hunk 3]
```diff
@@ -61,12 +61,12 @@ impl ELChainWriter {
             Some(metadata) => {
                 let contract_call =
                     contract_delegation_manager.registerAsOperator(op_details, metadata);
-                contract_call.gas(130000)
+                contract_call.gas(300000)
```
Comment (dralves) [inline crates/chainio/clients/elcontracts/src/writer.rs:64]: pull a constant with proper naming

File: crates/chainio/utils/src/lib.rs [hunk 4]
```diff
@@ -2,38 +2,4 @@
...
-    let x_1: BigInteger256 = g2.x.c1.into();
-    let y_0: BigInteger256 = g2.y.c0.into();
-    let y_1: BigInteger256 = g2.y.c1.into();
-
-    G2Point {
-        X: [biginteger256_to_u256(x_0), biginteger256_to_u256(x_1)],
-        Y: [biginteger256_to_u256(y_0), biginteger256_to_u256(y_1)],
-    }
-}
+// #![cfg_attr(not(test), warn(unused_crate_dependencies))]
```
Comment (dralves) [inline crates/chainio/utils/src/lib.rs:5]: unused? uncomment or remove

File: crates/crypto/bn254/src/lib.rs [hunk 5]
```diff
@@ -2,6 +2,6 @@
     html_logo_url = "https://github.com/Layr-Labs/eigensdk-rs/assets/91280922/bd13caec-3c00-4afc-839a-b83d2890beb5",
     issue_tracker_base_url = "https://github.com/Layr-Labs/eigensdk-rs/issues/"
 )]
-#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+// #![cfg_attr(not(test), warn(unused_crate_dependencies))]
```
Comment (dralves) [inline crates/crypto/bn254/src/lib.rs:5]: unused? uncomment or remove

File: crates/services/avsregistry/src/chaincaller.rs [hunk 6]
```diff
@@ -1,147 +1,125 @@
...
-use eigen_chainio_utils::convert_to_bn254_g1_point;
-use eigen_client_avsregistry::reader::AvsRegistryChainReader;
-use eigen_crypto_bls::attestation::G1Point as BlsG1Point;
-use eigen_crypto_bn254::utils::u256_to_bigint256;
-use eigen_logging::{logger::Logger, tracing_logger::TracingLogger};
-use eigen_services_operatorsinfo::operatorsinfo_inmemory::OperatorInfoServiceInMemory;
-use eigen_types::operator::{OperatorAvsState, OperatorInfo, OperatorPubKeys, QuorumAvsState};
-use eigen_utils::binding::BLSApkRegistry::G1Point;
-use std::collections::HashMap;
+// use alloy_primitives::{Bytes, FixedBytes, U256};
```
Comment (dralves) [inline crates/services/avsregistry/src/chaincaller.rs:1]: this needs cleanup

File: examples/avsregistry-write/examples/register_operator_in_quorum_with_avs_registry_coordinator.rs [hunk 7]
```diff
@@ -1,65 +1,82 @@
...
+use eigen_logging::get_test_logger;
+use eigen_testing_utils::m2_holesky_constants::{
+    AVS_DIRECTORY_ADDRESS, DELEGATION_MANAGER_ADDRESS, OPERATOR_STATE_RETRIEVER,
+    REGISTRY_COORDINATOR, SLASHER_ADDRESS, STRATEGY_MANAGER_ADDRESS,
+};
+use eigen_types::operator::Operator;
+use eyre::Result;
+use std::str::FromStr;
+#[tokio::main]
+async fn main() -> Result<()> {
```
Comment (dralves) [inline examples/avsregistry-write/examples/register_operator_in_quorum_with_avs_registry_coordinator.rs:18]: examples need pretty good comments

**Review by dralves — COMMENTED** [review]
still needs some cleanup

**Comment by supernovahs** [comment]
@dralves

**Review by dralves — APPROVED** [review]

**Pull request merged**